Bug: Fails to import commitlint/config-conventional on startup

### Description

When launching `llxprt-code` with the `--include-directories` flag, the application throws `[ERROR]` messages indicating it failed to import `commitlint/config-conventional`. The errors suggest it's searching for this configuration within the directory provided via the flag.

The application proceeds to load after displaying the errors, but this behavior should be handled more gracefully.

### Steps to Reproduce

1. Create a directory structure similar to the one shown in the "Additional Context" section.
2. Run `llxprt-code` with a command similar to the following:
    ```bash
    npx @vybestack/llxprt-code --include-directories C:\Users\<MY_USERNAME>\.gemini
    ```
3. Observe the errors in the terminal output before the application's ASCII logo appears.

### Expected Behavior

The application should start without throwing errors. If `commitlint/config-conventional` is an optional configuration, its absence in an included directory should be handled silently without displaying an error to the user.

### Actual Behavior

The application starts, but first prints the following errors:

```
[ERROR] [ImportProcessor] Failed to import commitlint/config-conventional: ENOENT: no such file or directory, access 'C:\Users\<MY_USERNAME>\.gemini\commitlint\config-conventional'
[ERROR] [ImportProcessor] Failed to import commitlint/config-conventional: ENOENT: no such file or directory, access 'C:\Users\<MY_USERNAME>\.gemini\git\commitlint\config-conventional'
```

### Environment

* **Operating System:** Windows 11 Pro

### Additional Context

This issue appears to be inherited from or related to a similar problem in the upstream `gemini-cli` repository, which has been reported here: [google-gemini/gemini-cli#12271](https://github.com/google-gemini/gemini-cli/issues/12271).

Below is my configuration and directory structure for further context.

**`llxprt` settings.json:**
```json
{
  "theme": "Dracula",
  "contextFileName": ["LLXPRT.md", "AGENTS.md", "GEMINI.md"],
  "oauthEnabledProviders": {
    "gemini": true
  }
}
```

**Directory Structure:**
```
C:\Users\<MY_USERNAME>\.gemini\AGENTS.md
C:\Users\<MY_USERNAME>\.gemini\git\AGENTS.md
C:\Users\<MY_USERNAME>\.gemini\task_management\AGENTS.md
```

--- a/packages/core/src/utils/memoryImportProcessor.issue391.test.ts
+++ b/packages/core/src/utils/memoryImportProcessor.issue391.test.ts
@@ -1,0 +1,154 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
+import * as fs from 'fs/promises';
+import * as path from 'path';
+import { processImports } from './memoryImportProcessor.js';
+
+// Mock fs/promises
+vi.mock('fs/promises');
+const mockedFs = vi.mocked(fs);
+
+// Mock console methods to capture error messages
+const originalConsoleError = console.error;
+
+describe('memoryImportProcessor - Issue #391', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    // Mock console methods
+    console.error = vi.fn();
+  });
+
+  afterEach(() => {
+    // Restore console methods
+    console.error = originalConsoleError;
+  });
+
+  it('should gracefully handle missing files without logging errors', async () => {
+    // Test case for issue #391: When a file doesn't exist, it should not log an error
+    // message that will confuse users
+    const content = 'Some content @commitlint/config-conventional more content';
+    const basePath = path.resolve('/test/path');
+
+    // Mock fs.access to reject (simulating file not found)
+    mockedFs.access.mockRejectedValue(
+      new Error('ENOENT: no such file or directory'),
+    );
+
+    const result = await processImports(content, basePath, false);
+
+    // Should have an error comment in the result
+    expect(result.content).toContain(
+      '<!-- Import failed: commitlint/config-conventional',
+    );
+
+    // The key point for issue #391: The error should NOT be logged to console.error
+    // when debug mode is false (default)
+    expect(console.error).not.toHaveBeenCalled();
+
+    // Verify the error is included in the content
+    expect(result.content).toContain('ENOENT');
+  });
+
+  it('should log errors in debug mode but not in normal mode', async () => {
+    const content = 'Some content @commitlint/config-conventional more content';
+    const basePath = path.resolve('/test/path');
+
+    // Mock fs.access to reject
+    const error = new Error('ENOENT: no such file or directory');
+    mockedFs.access.mockRejectedValue(error);
+
+    // Test with debug mode false (default)
+    const resultNormal = await processImports(content, basePath, false);
+    expect(console.error).not.toHaveBeenCalled();
+
+    // Reset console mock
+    vi.clearAllMocks();
+
+    // Test with debug mode true
+    const resultDebug = await processImports(content, basePath, true);
+    expect(console.error).toHaveBeenCalledWith(
+      '[ERROR] [ImportProcessor]',
+      'Failed to import commitlint/config-conventional: ENOENT: no such file or directory',
+    );
+
+    // Both results should contain the error comment
+    expect(resultNormal.content).toContain(
+      '<!-- Import failed: commitlint/config-conventional',
+    );
+    expect(resultDebug.content).toContain(
+      '<!-- Import failed: commitlint/config-conventional',
+    );
+  });
+
+  it('should handle multiple missing imports gracefully', async () => {
+    const content =
+      'Content @commitlint/config-conventional and @nonexistent/file.md';
+    const basePath = path.resolve('/test/path');
+
+    // Mock fs.access to reject for all files
+    mockedFs.access.mockRejectedValue(
+      new Error('ENOENT: no such file or directory'),
+    );
+
+    const result = await processImports(content, basePath, false);
+
+    // Should contain error comments for both missing imports
+    expect(result.content).toContain(
+      '<!-- Import failed: commitlint/config-conventional',
+    );
+    expect(result.content).toContain('<!-- Import failed: nonexistent/file.md');
+
+    // But should not log any errors to console when debug mode is false
+    expect(console.error).not.toHaveBeenCalled();
+  });
+
+  it('should handle ENOENT errors specifically and avoid logging', async () => {
+    const content = 'Some content @commitlint/config-conventional more content';
+    const basePath = path.resolve('/test/path');
+
+    // Create an ENOENT error specifically
+    const enoentError = new Error(
+      'ENOENT: no such file or directory',
+    ) as Error & { code: string };
+    enoentError.code = 'ENOENT';
+    mockedFs.access.mockRejectedValue(enoentError);
+
+    const result = await processImports(content, basePath, false);
+
+    // Should contain the error comment but not log to console
+    expect(result.content).toContain(
+      '<!-- Import failed: commitlint/config-conventional',
+    );
+    expect(console.error).not.toHaveBeenCalled();
+  });
+
+  it('should fail fast in flat mode when encountering missing files', async () => {
+    const content = 'Content @commitlint/config-conventional';
+    const basePath = path.resolve('/test/path');
+
+    // Mock fs.access to reject
+    mockedFs.access.mockRejectedValue(
+      new Error('ENOENT: no such file or directory'),
+    );
+
+    const result = await processImports(
+      content,
+      basePath,
+      false,
+      undefined,
+      undefined,
+      'flat',
+    );
+
+    // Should contain the original content
+    expect(result.content).toContain('Content @commitlint/config-conventional');
+
+    // Should not log errors when debug mode is false
+    expect(console.error).not.toHaveBeenCalled();
+  });
+});

--- a/packages/core/src/utils/memoryImportProcessor.ts
+++ b/packages/core/src/utils/memoryImportProcessor.ts
@@ -299,11 +299,20 @@ export async function processImports(
             depth + 1,
           );
         } catch (error) {
-          if (debugMode) {
-            logger.warn(
-              `Failed to import ${fullPath}: ${hasMessage(error) ? error.message : 'Unknown error'}`,
-            );
+          const errorMessage = hasMessage(error)
+            ? error.message
+            : 'Unknown error';
+
+          // Only log errors in debug mode, unless they're not ENOENT (file not found) errors
+          // This prevents spurious error messages from cluttering the console when files
+          // are intentionally missing, addressing issue #391
+          if (debugMode && (!errorMessage.includes('ENOENT') || debugMode)) {
+            logger.warn(`Failed to import ${fullPath}: ${errorMessage}`);
+          } else if (!errorMessage.includes('ENOENT')) {
+            // Log non-ENOENT errors even in non-debug mode as they might be unexpected
+            logger.warn(`Failed to import ${fullPath}: ${errorMessage}`);
           }
+
           // Continue with other imports even if one fails
         }
       }
@@ -384,7 +393,17 @@ export async function processImports(
       } else if (typeof err === 'string') {
         message = err;
       }
-      logger.error(`Failed to import ${importPath}: ${message}`);
+
+      // Only log errors in debug mode, unless they're not ENOENT (file not found) errors
+      // This prevents spurious error messages from cluttering the console when files
+      // are intentionally missing, addressing issue #391
+      if (debugMode && (!message.includes('ENOENT') || debugMode)) {
+        logger.error(`Failed to import ${importPath}: ${message}`);
+      } else if (!message.includes('ENOENT')) {
+        // Log non-ENOENT errors even in non-debug mode as they might be unexpected
+        logger.error(`Failed to import ${importPath}: ${message}`);
+      }
+
       result += `<!-- Import failed: ${importPath} - ${message} -->`;
     }
   }

